Sources/PackageModel/Module.swift

```diff
@@ -51,13 +51,13 @@ public enum ModuleType {
 public protocol ModuleTypeProtocol {
     var sources: Sources { get }
     var type: ModuleType { get }
-    var mainFile: String { get }
+    var mainFiles: Set<String> { get }
 }
 
 extension ModuleTypeProtocol {
     public var type: ModuleType {
         let isLibrary = !sources.relativePaths.contains { path in
-            path.basename.lowercased() == mainFile
+            mainFiles.contains(path.basename.lowercased())
```

Comment:
I wonder if we could just look for a file in the sources named "main.<suffix>" instead of needing to duplicate the list of all types?